specs/beacon-chain.md

```diff
@@ -324,36 +324,39 @@ def shuffle(values: List[Any],
     """
     values_count = len(values)
 
-    # entropy is consumed in 3 byte chunks
-    # sample_max is defined to remove the modulo bias from this entropy source
-    sample_max = 2 ** 24
-    assert values_count <= sample_max
+    # Entropy is consumed from the seed in 3-byte (24 bit) chunks.
+    rand_max = 2 ** 24 - 1
+    assert values_count < rand_max
 
     output = [x for x in values]
     source = seed
     index = 0
-    while index < values_count:
+    while index < values_count - 1:
         # Re-hash the source
         source = hash(source)
-        for position in range(0, 30, 3):  # gets indices 3 bytes at a time
```

Comment:
The number `30` is not immediately obvious. Consider replacing by `31` (the max index) `- 3` (the constant above).

Reply:
Agreed

Reply:
Confusing either way imo 😬

Reply:
This is interesting.. I suspect it should be `range(0, 32, 3)` and we're throwing away 3 bytes.

Will investigate and report back.

Reply:
However, that's only sensible when `rand_bytes % seed_bytes != 0`.

Will think on this some more after the call.

Reply:
that works for me.